### PR TITLE
3 API calls

### PR DESCRIPTION
**Pull request by statu**
when entering a city name, "minneapolis", current weather data populates the page. When entering city code "5037649" (minneapolis), current weather and 5 day fore populate the page. Two calls when entering city code and one call when entering city name. 3 total. Does this suffice?

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+                weatherApp.getWeatherDataCurrentName(city);
+            }
+            
+
+            console.log(apikey);
+            //console.log(city);
+        },
+        
+        getWeatherDataCurrent : function (citycode) {
+            var urlCurrent = "http://api.openweathermap.org/data/2.5/weather?id=" + citycode + "appid=" + weatherApp.weatherApiKey + "&units=imperial";
```
Comment (mbMosman) [inline js/script.js]: One API call...

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+             getFormDataCurrentName: function (city) {
+            if (weatherApp.weatherApiKey === null || weatherApp.weatherApiKey === "") {
+                weatherApp.weatherApiKey = $("#apikey").val().trim();
+                weatherApp.saveAPIKey();
+            }
+            var cityname = $("#city").val().trim();
+            if (cityname === null) {
+                weatherApp.$targetArea.html("Enter a city.");
+            } else {
+                weatherApp.getWeatherDataCurrentName(city);
```
Comment (mbMosman) [inline js/script.js]: The form uses only the getWeatherDataCurrentName method...

File: js/script.js [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        },
+        
+    };
+
+    $("#submit").click(function () {
+        var city = $("#city").val().trim();
+                
+        if (isNaN(city))
+            {
+                weatherApp.getFormDataCurrentName(city);
```
Comment (mbMosman) [inline js/script.js]: Ah! I see... if you enter a code it tries to do it by code... You might add that on the form page or something.

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+                    weatherApp.$targetArea.html("Sorry, no weather data available. Try again later.");
+                }
+            }).fail ( function() {
+                
+                weatherApp.$targetArea.html("Sorry, no weather data available. Try again later.");
+            });
+            
+        },
+
+        loadApiKey: function () {
```
Comment (mbMosman) [inline js/script.js]: Why do you have more than one load and save API key function?

File: js/script.js [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
```
Comment (mbMosman) [inline js/script.js]: These are all the same, right?

File: js/script.js (hunk 4 above)
Comment (statu) [inline js/script.js]: I wanted to make sure it was just going to work before I tried deleting duplicate stuff. 

Sean Tatu

**Comment by mbMosman** [comment]
Great job with this and the video.

**Pull request closed without merging**